client-sdk/go/client/client.go

```diff
@@ -318,11 +342,99 @@ func (rc *runtimeClient) GetTransactionsWithResults(ctx context.Context, round u
 }
 
 // Implements RuntimeClient.
-func (rc *runtimeClient) GetEvents(ctx context.Context, round uint64) ([]*coreClient.Event, error) {
-	return rc.cc.GetEvents(ctx, &coreClient.GetEventsRequest{
+func (rc *runtimeClient) GetEventsRaw(ctx context.Context, round uint64) ([]*types.Event, error) {
+	rawEvs, err := rc.cc.GetEvents(ctx, &coreClient.GetEventsRequest{
+		RuntimeID: rc.runtimeID,
+		Round:     round,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	evs := make([]*types.Event, len(rawEvs))
+	for i, rawEv := range rawEvs {
+		var ev types.Event
+		if err := ev.UnmarshalRaw(rawEv.Key, rawEv.Value); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal event '%v': %w", rawEv, err)
+		}
+		evs[i] = &ev
+	}
+
+	return evs, nil
+}
+
+// Implements RuntimeClient.
+func (rc *runtimeClient) GetEvents(ctx context.Context, round uint64, decoders []EventDecoder, includeUndecoded bool) ([]DecodedEvent, error) {
+	rawEvs, err := rc.cc.GetEvents(ctx, &coreClient.GetEventsRequest{
 		RuntimeID: rc.runtimeID,
 		Round:     round,
 	})
+	if err != nil {
+		return nil, err
+	}
+
+	evs := make([]DecodedEvent, 0)
+OUTER:
+	for _, rawEv := range rawEvs {
+		var ev types.Event
+		if err := ev.UnmarshalRaw(rawEv.Key, rawEv.Value); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal event '%v': %w", rawEv, err)
+		}
+		for _, decoder := range decoders {
+			decoded, err := decoder.DecodeEvent(&ev)
+			if err != nil {
+				return nil, fmt.Errorf("failed to decode event: %w", err)
+			}
+			if decoder != nil {
+				evs = append(evs, decoded)
+				continue OUTER
+			}
+		}
+		if includeUndecoded {
+			evs = append(evs, &ev)
+		}
+	}
+
+	return evs, nil
+}
+
+// Implements RuntimeClient.
+func (rc *runtimeClient) WatchEvents(ctx context.Context, decoders []EventDecoder, includeUndecoded bool) (<-chan *BlockEvents, pubsub.ClosableSubscription, error) {
+	notifier := pubsub.NewBroker(true)
+	typedCh := make(chan *BlockEvents)
+	sub := notifier.Subscribe()
+	sub.Unwrap(typedCh)
```

Comment:
Why are you creating a new broker here if you only have one subscription? A channel should be enough.

Reply:
fixed